### DISCUSSION TITLE
Accesso Account google

### DISCUSSION THREAD

**Discussion by sempliciniluigi89** [iisgiua/giuaschool#164] [Richieste di aiuto] [answered]
Inoltre , non avendo una guida chiara e precisa non so con quali dati compilare i seguenti campi presenti nel file env. ossia, autenticazione tramite Google Workspace
GOOGLE_API_KEY=''
GOOGLE_CLIENT_ID=''
GOOGLE_CLIENT_SECRET=''
OAUTH_GOOGLE_CLIENT_ID=''
OAUTH_GOOGLE_CLIENT_SECRET=''
OAUTH_GOOGLE_CLIENT_HD=''

Si deve pagare qualcosa per ottenere queste credenziali che permettono l'accesso direttamente con l'account?Grazie

**Comment by trinko** [comment, marked as answer]
Se avete Google Workspace, anche in versione Educational (quella gratuita), è possibile usare gli utenti google. 
Però la configurazione richiesta sui sistemi Google non è per niente semplice.

Inoltre, una volta fatto tutto, il registro consentirà la sola autenticazione tramite Google. 
Le funzioni di creazione automatica dell'utente Google quando si crea un utente del registro, di modifica password e altre che gestiscono i corsi Classroom, non potranno essere usate.
Questo perché vengono attivate da una procedura temporizzata sul server (cron), che non è disponibile nei servizi di hosting più semplici (tipo altervista).

**Reply by MatteoGheza** [reply]
> Questo perché vengono attivate da una procedura temporizzata sul server (cron), che non è disponibile nei servizi di hosting più semplici (tipo altervista).

@trinko per quanto riguarda l'utilizzo di Cron su hosting che non lo permettono (per esempio Altervista), consiglio l'uso di https://cron-job.org/en/ (cron-job.org).
È gratuito e open source, permettendo self-hosting.
Non ha limiti sul numero di Cronjobs che si possono aggiungere (se non ricordo male) e permette di eseguirli ogni minuto, inoltre, permette di conservare le risposte ottenute dal server e gli status code delle risposte per poterli visualizzare in una dashboard.

Ha anche funzioni più avanzate, come l'uso di REST API per la modifica delle impostazioni, e la realizzazione di "status page" aggiornate in base alle risposte (status code delle risposte e tempi) dei Cronjobs aggiunti.

Lo uso in produzione dal 2018 (per una webapp hostata su hosting economico Aruba che permette di eseguire i Cronjobs solo ogni ora, mentre a me serviva eseguirli ogni 5 minuti) con 3 Cronjobs, e non ho mai avuto problemi.

Allego screenshots della pagina gestionale di questo programma.
![IMG_20220208_080206](https://user-images.githubusercontent.com/27729549/152936930-e3480cfc-bbb4-4e5c-a9c9-0e27c58c4208.jpg)
![IMG_20220208_080324](https://user-images.githubusercontent.com/27729549/152936969-f2f0ac2d-eeca-4c80-abe8-8005baf7718d.jpg)
![IMG_20220208_080542](https://user-images.githubusercontent.com/27729549/152936991-637b9016-a4da-4387-869c-53e57623cf0c.jpg)
![IMG_20220208_080652](https://user-images.githubusercontent.com/27729549/152937011-7e7ab4ab-e7c0-4087-af0c-607b7ce0a4b2.jpg)
![IMG_20220208_080759](https://user-images.githubusercontent.com/27729549/152937067-31d0d99e-2eed-4c64-aa48-ace1d499923c.jpg)

**Comment by sempliciniluigi89** [comment]
Scusate l'ignoranza, ma non ho capito di cosa si tratta e a cosa serve

Il Mar 8 Feb 2022, 08:19 Matteo Gheza ***@***.***> ha scritto:
